subworkflows/ebi-metagenomics/assembly_decontamination/tests/main.nf.test

```diff
@@ -0,0 +1,36 @@
+nextflow_workflow {
+    
+    name "Test subworkflow ASSEMBLY_DECONTAMINATION"
+    script "../main.nf"
+    config "./nextflow.config"
+    workflow "ASSEMBLY_DECONTAMINATION"
+
+    tag "subworkflows"
+    tag "subworkflows_nfcore"
+    tag "subworkflows/assembly_decontamination"
+    tag "blast/blastn"
+    tag "seqkit/grep"
+
+    test("Assembly host decontamination") {
+        when {
+            params {
+                blast_reference_genomes_folder    = "${moduleDir}/tests/data"
+            }
+            workflow {
+                """
+                input[0] = Channel.of([
+                    [ id: 'test' ], // meta map
+                    file("${moduleDir}/tests/data/test.fa", checkIfExists: true)
+                ])
+                input[1] = "contamination.fa"
+                """
+            }
+        }
+        then {
+            assertAll(
+                { assert workflow.success },
+                { assert snapshot(workflow.out.cleaned_contigs).match() }
```

Comment:
I think it's worth it to check which contigs are being returned here. There is a plugin for nf-test that allows to do assertions on fasta files (https://www.nf-test.com/docs/assertions/fasta/)

I think in your case it would look like:

```
with(workflow.out.cleaned_contigs) {
  def clean_contigs = path(get(0)).fasta
  assert "uncontaminated_contig" in clean_contigs
  assert !("partially_contaminated_contig" in sequences)
  assert !("contaminated_contig" in sequences)
}
```